plugins/com.google.cloud.tools.eclipse.ui.util/src/com/google/cloud/tools/eclipse/ui/util/databinding/ProjectSelectorValidator.java

```diff
@@ -17,36 +17,27 @@
 package com.google.cloud.tools.eclipse.ui.util.databinding;
 
 import com.google.cloud.tools.eclipse.ui.util.Messages;
-import com.google.cloud.tools.project.ProjectIdValidator;
+import com.google.common.base.Preconditions;
 import org.eclipse.core.databinding.validation.IValidator;
 import org.eclipse.core.databinding.validation.ValidationStatus;
 import org.eclipse.core.runtime.IStatus;
 
-public class ProjectIdInputValidator implements IValidator {
+public class ProjectSelectorValidator implements IValidator {
   private boolean requireProjectId = true;
 
-  public ProjectIdInputValidator(boolean requireProjectId) {
+  public ProjectSelectorValidator(boolean requireProjectId) {
     this.requireProjectId = requireProjectId;
   }
 
   @Override
   public IStatus validate(Object input) {
-    if (!(input instanceof String)) {
-      return ValidationStatus.error(Messages.getString("project.id.invalid")); //$NON-NLS-1$
-    }
-    String value = (String) input;
-    return validateString(value);
-  }
+    Preconditions.checkState(input == null || input instanceof String);
```

Comment:
Should a non-null non-String parameter be a validation error (i.e. error status returned) instead of an IllegalArgumentException?